### ISSUE TITLE
Detail work to validate feeds!

### ISSUE THREAD

**Issue by vsoch** [hpc-social/events#1]
Currently - I have rough drafts of feeds with (known) missing fields or otherwise invalid. I just need to fix these (and then I think they should show up, and the ics file / ical feed should be usable. Just ran out of steam today :)

**Comment by alansill** [comment]
I tried subscribing to the overall events listing in my Apple Calendar using the ical URL. I see multiple instances of each event listed, not sure why.
<img width="1798" alt="Screenshot 2023-01-26 at 4 00 08 PM" src="https://user-images.githubusercontent.com/2364484/214960808-5768a2b2-253d-4bbb-8a9c-1f3b3b439c7c.png">
<img width="1795" alt="Screenshot 2023-01-26 at 4 01 50 PM" src="https://user-images.githubusercontent.com/2364484/214960829-bb7955ae-a6da-4826-ad71-10f1dc4494b0.png">

**Comment by alansill** [comment]
BTW If we are going to have more than one choice of calendar event type we should probably separate them completely from each other so that people can decide which one(s) to subscribe to without potentially duplicating events if they subscribe to more than one calendar. (In the above case I only subscribed to the overall events one to keep things clean for testing.)

**Comment by vsoch** [comment]
Yeah I'm not sure how to best debug this - when I tried adding to Google calendar I didn't see it show up at all. I was hoping someone else can take a look I'm quite busy atm.

**Comment by vsoch** [comment]
As a sanity check I'd look at the feed file and see that we aren't actually adding it multiple times!

**Comment by alansill** [comment]
Yes, that's certainly the problem. We'll need to deduplicate the entries somehow and/or allow for editing of the list.

Presumably we're going to embed a calendar for display in the page for this on the web site. I can make one in our hpc.social GSuite for this purpose if needed, associated with some utility account, and we'll need to set up a subscription in that for the entries. Need to figure out the cause-and-effect -- is the idea that the list from the form would be turned into a susription link that anyone could add, or that it would be fed to a GSuite-based calendar that people could then use to subscribe to rather than the direct output from the form? Having an actual calendar -- perhaps there's a shared calendar feature in Google Calendar we could set up, I'll have to look into that -- could provide an intermediate point for editing perhaps and then people could subscribe to that calendar.

**Comment by vsoch** [comment]
@alansill I generate that file programmatically, if there are repeats there is a bug! Can you share a screenshot please?

**Comment by vsoch** [comment]
And @alansill we already have a calendar that receives this feed, I guess you haven't looked :) https://hpc.social/events/

**Comment by alansill** [comment]
> @alansill I generate that file programmatically, if there are repeats there is a bug! Can you share a screenshot please?

$ grep -n -C2 NAME:Container all.ical 
29- ork
30-LOCATION:virtual
31:NAME:Container Plumbing 2023
32-END:VEVENT
33-BEGIN:VEVENT
--
54- ork
55-LOCATION:virtual
56:NAME:Container Plumbing 2023
57-END:VEVENT
58-BEGIN:VEVENT
--
79- ork
80-LOCATION:virtual
81:NAME:Container Plumbing 2023
82-END:VEVENT
83-BEGIN:VEVENT
--
104- ork
105-LOCATION:virtual
106:NAME:Container Plumbing 2023
107-END:VEVENT
108-BEGIN:VEVENT
--
129- ork
130-LOCATION:virtual
131:NAME:Container Plumbing 2023
132-END:VEVENT
133-BEGIN:VEVENT
--
154- ork
155-LOCATION:virtual
156:NAME:Container Plumbing 2023
157-END:VEVENT
158-BEGIN:VEVENT
--
179- ork
180-LOCATION:virtual
181:NAME:Container Plumbing 2023
182-END:VEVENT
183-END:VCALENDAR

**Comment by vsoch** [comment]
Well I put a lot of work into getting us most of the way there with the web interface - that was non-trivial to figure out. And the above just looks like a bug with figuring out when an event is repeated, which I'm actually fairly happy about.

**Comment by vsoch** [comment]
I haven't made any kind of announcement, so you can consider it still in testing. I was hoping someone would be able to help out, but maybe not.

**Comment by vsoch** [comment]
okay just pushed a fairly simple fix - instead of reading in an existing file, we just re-generate the whole thing each time.

**Comment by alansill** [comment]
Thanks. Swamped this week-this is the best I could manage.

**Comment by vsoch** [comment]
yeah no worries! Let's pick up on in the not too distant future when stuffs dies down :)

**Comment by vsoch** [comment]
Good to close here?
